src/main/java/ava/processor/Parser.java

```diff
@@ -0,0 +1,217 @@
+                }
+
+            case EVENT:
+                String subStringEvent = chat.substring(6);
+                if (subStringEvent.split(" /at ").length == 1) {
+                    throw new NoTimeException(command.name());
+                } else {
+                    try {
+                        return new Event(subStringEvent.split(" /at ")[0], false,
+                                LocalDateTime.parse(subStringEvent.split(" /at ")[1], TIME_FORMAT));
+                    } catch (Exception e) {
+                        throw new WrongTimeFormatException();
+                    }
+                }
+
+            default:
+                throw new NoCommandException(command.name());
+            }
+        } else {
+            switch (command) {
+            case TODO:
+                throw new NoDescriptionException(command.name());
+            case DEADLINE:
+                throw new NoDescriptionException(command.name());
+            case EVENT:
+                throw new NoDescriptionException(command.name());
+            default:
+                throw new NoCommandException(command.name());
+            }
+        }
+    }
+
+    /**
+     * Returns a new delete task of the specified current task.
+     *
+     * @param chat Input from the scanner.
+     * @param tasklist ArrayList of tasks.
+     * @return Task object
+     * @throws NoDescriptionException If there are no description.
+     * @throws NoCommandException If there are no commands.
+     */
+    public static Task delete(String chat, TaskList tasklist) throws NoDescriptionException {
+
+        int order = tasklist.size();
+        if (chat.split(" ").length == 1) {
+            throw new NoDescriptionException("Delete");
+        } else {
+            int num = Integer.parseInt(chat.split(" ")[1]) - 1;
+            return new Delete(num);
+        }
+    }
+
+    /**
+     * Returns a new find task to find task with the matching keyword
+     *
+     * @param chat Input from the scanner.
+     * @param tasklist ArrayList of tasks.
+     * @return Find object
+     * @throws NoDescriptionException If there are no description.
+     */
+    public static Find find(String chat, TaskList tasklist) throws NoDescriptionException {
+        if (chat.split(" ").length == 1) {
+            throw new NoDescriptionException("Find");
+        } else {
+            return new Find(chat.substring(5));
+        }
+    }
+
+    /**
+     * Parses the input and returns the task of the specified command.
+     *
+     * @param chat Input from the scanner.
+     * @param tasklist ArrayList of tasks.
+     * @return Task object.
+     * @throws NoCommandException If there are no commands.
+     */
+    public static Task parse(String chat, TaskList tasklist) throws NoCommandException {
+
+        Commands command;
+
+        try {
+            try {
+                command = Parser.Commands.valueOf(chat.toUpperCase().split(" ")[0]);
+            } catch (Exception e) {
```

Comment:
Perhaps this exception could be further specified as an IllegalArgumentException? I noticed this recurrence of catching Exception, instead of a more specific type, in other places too.